thunder/clang/__init__.py

```diff
@@ -715,8 +731,49 @@ def _advanced_indexing(a: TensorLike, /, key) -> TensorLike:
         lambda: f"Advanced indexing currently only supports keys that are ellipses, integer tensors or sequences, but got {key=}",
     )
 
+    def _to_tensorproxies(x: list, device: devices.DeviceType):
+        # convert list to tensor
+        # e.g. [1, 2] -> tensor.Tensor([1, 2])
+        for idx, val in enumerate(x):
+            if isinstance(val, (NumberProxy)):
+                x[idx] = utils.get_numberlike_value(val)
+        if all(isinstance(i, int) for i in x):
+            x = tensor_from_sequence(x, dtype=dtypes.int64, device=device)
+        return x
+
+    basic_keys = []  # (key index, key)
+    advanced_keys = []  # (key index, key)
+
+    # convert list to 1D tensor
+    # this handles both cases when key is list or a tuple that contains lists
+    if isinstance(key, Sequence):
+        if isinstance(key, list):
+            key = _to_tensorproxies(key, a.device)
+            if isinstance(key, list):
+                # handle list of tensors
+                for ii, i in enumerate(key):
+                    advanced_keys.append((ii, i))
+        else:
+            key_ = []
+            for key_idx, x in enumerate(key):
+                if isinstance(x, list):
```

Comment:
Wouldn't this check be any for any Sequence, not just list?

Reply:
I would like to understand what other Sequence we want to cover in advanced indexing. Would you be able to explain that? Or provide me an example?

Reply:
Let me update this in a follow up PR that can handle more advanced indexing!

Reply:
Following the NumPy documentation, I think any Sequence is something that we should attempt to convert to a tensor?